tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,724 @@
+    private static final String NAME_ARG = "name";
+    private static final String DEFAULT_FILESYSTEM_REALM_NAME = "encrypted-filesystem-realm";
+    public static Supplier<Provider[]> ELYTRON_PASSWORD_PROVIDERS = () -> new Provider[]{
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+
+    private final List<Descriptor> descriptors = new ArrayList<>();
+    private final List<String> PARAMS_LIST = new ArrayList<>(Arrays.asList(INPUT_REALM_LOCATION_PARAM, OUTPUT_REALM_LOCATION_PARAM));
+
+    private final Options options;
+    private final CommandLineParser parser = new DefaultParser();
+    private CommandLine cmdLine = null;
+    private boolean silentMode = false;
+    private boolean summaryMode = false;
+    private StringBuilder summaryString = null;
+    private boolean warningOccurred = false;
+
+    FileSystemEncryptRealmCommand() {
+        options = new Options();
+        Option option;
+
+        option = new Option("i", INPUT_REALM_LOCATION_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptInputLocationDesc());
+        option.setArgName(DIRECTORY_ARG);
+        options.addOption(option);
+
+        option = new Option("r", REALM_NAME_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptNewRealmDesc());
+        option.setArgName(DIRECTORY_ARG);
+        options.addOption(option);
+
+        option = new Option("o", OUTPUT_REALM_LOCATION_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptOutputLocationDesc());
+        option.setArgName(DIRECTORY_ARG);
+        options.addOption(option);
+
+        option = new Option("c", CREDENTIAL_STORE_LOCATION_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptCredentialStoreDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("a", CREATE_CREDENTIAL_STORE_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptCreateCredentialStoreDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+
+        option = new Option("s", SECRET_KEY_ALIAS_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptSecretKeyDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("e", HASH_ENCODING_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptHashEncodingDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("f", ENCODED_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptEncodedDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("l", LEVELS_PARAM, true, ElytronToolMessages.msg.cmdFileSystemEncryptLevelsDesc());
+        option.setArgName(FILE_ARG);
+        options.addOption(option);
+
+        option = new Option("b", BULK_CONVERT_PARAM, true, ElytronToolMessages.msg.cmdFileSystemRealmEncryptBulkConvertDesc());
+        option.setArgName(NAME_ARG);
```

Comment:
@Ashpan  just minor, I think this should be `option.setArgName(FILE_ARG);` since the file path to the descriptor file should be provided